Microsoft Entra ID Content Pack (#35379)
demisto/content: Packs/MicrosoftEntraID/README.md

<~XSIAM>  

<html>

<h1>Microsoft Entra ID</h1>

<details>
    <summary><h2 style="display:inline-block">What does this pack do</h2></summary>

<h3>Log Normalization - One Data Model</h3>

> <p>This pack support normalization of the below log categories of Microsoft Entra ID:</p>

Reviewer: ```suggestion
<p>This pack supports normalization of the below log categories of Microsoft Entra ID:</p>
```

> <ol>

Reviewer: This should be a bulleted list.  Numbered lists are used when you have to do something in a particular order such as instructions.

Author: Changed it.

    <li>AuditLogs</li>
    <li>SignInLogs</li>
    <li>NonInteractiveUserSignInLogs</li>
    <li>ServicePrincipalSignInLogs</li>
    <li>ManagedIdentitySignInLogs</li>
    <li>ADFSSignInLogs</li>
    <li>ProvisioningLogs</li>
    <li>RiskyUsers</li>
    <li>UserRiskEvents</li>
    <li>RiskyServicePrincipals</li>
    <li>ServicePrincipalRiskEvents</li>
</ol>

<h3>Timestamp Parsing</h3>

> <p>Timestamp parsing relies on 2 fields, which depends on the log category:</p>

Reviewer: Should this be 3 fields?

Author: You are right.

> <ol>

Reviewer: These should also be a bulleted list.

Author: changed.

  <li>`properties.createdDateTime`
    <ol>
        <li>SignInLogs</li>
        <li>NonInteractiveUserSignInLogs</li>
        <li>ServicePrincipalSignInLogs</li>
        <li>ManagedIdentitySignInLogs</li>
        <li>ADFSSignInLogs</li>
    </ol>
  </li>
  <li>`properties.activityDateTime`
    <ol>
        <li>AuditLogs</li>
        <li>ProvisioningLogs</li>
        <li>UserRiskEvents</li>
        <li>ServicePrincipalRiskEvents</li>
    </ol>
  </li>
  <li>`properties.riskLastUpdatedDateTime`
    <ol>
        <li>RiskyUsers</li>
        <li>RiskyServicePrincipals</li>
    </ol>
  </li>
</ol>
</details>
<hr>

<details>
<summary><h2 style="display:inline-block">Data Collection</h2></summary>
<h3 >Entra ID Side</h3>

> <p>To configure Microsoft Entra ID to send logs to XSIAM, follow the below steps.</p>

Reviewer: ```suggestion
<p>To configure Microsoft Entra ID to send logs to Cortex XSIAM, follow the below steps.</p>
```

<h4>Prerequisites</h4>
<ol>

>     <li>Create an <b>Azure event hub</b>. For more information, refer to Microsoft's official <a href="https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create">documetaion</a>.</li>

Reviewer: ```suggestion
    <li>Create an <b>Azure event hub</b>. For more information, refer to Microsoft's official <a href="https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create">documentation</a>.</li>
```

    <li>Make sure that you have at least Security Administrator role.</li>
</ol>
<h4>Stream logs to an event hub</h4>
<ol>
    <li>Sign in to the <b>Microsoft Entra admin center</b>.</li>
    <li>Navigate to <b>Identity</b> &rarr; <b>Monitoring & health</b> &rarr; <b>Diagnostic settings</b>.</li>
    <li>Select <b>+ Add diagnostic setting</b> to create a new integration or select <b>Edit setting</b> for an existing integration.</li>
    <li>Enter a <b>Diagnostic setting name</b>. If you're editing an existing integration, you can't change the name.</li>
    <li>Select the log categories that you want to stream. Refer to the <b>Log Normalization</b> section for the supported log categories for normalization.</li>

>     <li>Select the Stream to an event hub check box.</li>

Reviewer: ```suggestion
    <li>Select the <b>Stream to an event hub </b>checkbox.</li>
```

    <li>Select the Azure subscription, Event Hubs namespace, and optional event hub where you want to route the logs.</li>
</ol>

> <p>For more information, refer to Microsoft's official <a href="https://learn.microsoft.com/en-us/entra/identity/monitoring-health/howto-stream-logs-to-event-hub">documetaion</a>.</p>

Reviewer: ```suggestion
<p>For more information, refer to Microsoft's official <a href="https://learn.microsoft.com/en-us/entra/identity/monitoring-health/howto-stream-logs-to-event-hub">documentation</a>.</p>
```

> <h3>XSIAM side</h3>

Reviewer: ```suggestion
<h3>Cortex XSIAM side</h3>
```

> <p>To connect XSIAM to the Azure Event Hub, follow the below steps.</p>

Reviewer: ```suggestion
<p>To connect Cortex XSIAM to the Azure Event Hub, follow the below steps.</p>
```

- RiskyServicePrincipals
- ServicePrincipalRiskEvents

### Supported Timestamp Formats:
- YYYY-MM-DDTHH:MM:SS.S* (UTC)
- YYYY-MM-DDTH:M:S.S* (UTC)

***

## Data Collection
To configure Microsoft Entra ID to send logs to Cortex XSIAM, follow the below steps.

### Prerequisites
- Create an **Azure event hub**. For more information, refer to Microsoft's official [documentation](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create).

> - Make sure that you have at least Security Administrator role.

Reviewer 2: at least *a* security administrator role

### Stream logs to an event hub

> 1. Sign in to the **Microsoft Entra admin center**.

Reviewer 2: Capitalize Admin Center

Author: It is with small letters in both ms docs and the portal itself.

> 2. Navigate to **Identity** &rarr; **Monitoring & health** &rarr; **Diagnostic settings**.

Reviewer 2: Capitalize Health and Settings

Author: It is with small letters in both ms docs and the portal itslef.

> 3. Select **+ Add diagnostic setting** to create a new integration or select **Edit setting** for an existing integration.

Reviewer 2: Capitalize if the options are capitalized in the official page

Author: Not capitalized in the official page

4. Enter a **Diagnostic setting name**. If you're editing an existing integration, you can't change the name.
5. Select the log categories that you want to stream. Refer to the **Log Normalization** section for the supported log categories for normalization.
6. Select the **Stream to an event hub** checkbox.
7. Select the Azure subscription, Event Hubs namespace, and optional event hub where you want to route the logs.

For more information, refer to Microsoft's official [documentation](https://learn.microsoft.com/en-us/azure/event-hubs/event-hubs-create">documentation).

### Cortex XSIAM side
To connect Cortex XSIAM to the Azure Event Hub, follow the below steps.

#### Azure Event Hub Collector
1. Navigate to **Settings** &rarr; **Data Sources**.

> 2. If you have already configured an **Azure Event Hub Collector**, select the **3 dots**, and then select **+ Add New Instance**. If not, select **+ Add Data Source**, search for "Azure Event Hub" and then select **Connect**.

Reviewer 2: Maybe turn "3 dots" to "...", or rather "select the '...' symbol"?

Author: Vertical dots.
